experimental/hip/native_executable.h

```diff
@@ -0,0 +1,50 @@
+// Copyright 2023 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#ifndef IREE_EXPERIMENTAL_HIP_NATIVE_EXECUTABLE_H_
+#define IREE_EXPERIMENTAL_HIP_NATIVE_EXECUTABLE_H_
+
+#include <stdint.h>
+
+#include "experimental/hip/dynamic_symbols.h"
+#include "experimental/hip/hip_headers.h"
+#include "iree/base/api.h"
+#include "iree/base/tracing.h"
+#include "iree/hal/api.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif  // __cplusplus
+
+typedef struct iree_hal_hip_kernel_params_t {
```

Comment:
I'd name this as `iree_hal_hip_kernel_info_t`--kernel parameters indicates the parameter/arguments to the kernel to me; so it can be confusing.